client/src/modules/depots/depots.js

```diff
@@ -1,154 +1,106 @@
-  vm.refreshValidation = refreshValidation;
-  vm.maxLength = util.maxTextLength;
-
-  /** Load depots */
-  depotsList();
-
-  function depotsList() {
-    DepotService.read()
-    .then(function (list) {
-      vm.depotList = list;
-    });
-  }
-
-  function update(uuid) {
-    initialise('action', 'update', uuid);
-  }
-
-  function create() {
-    initialise('action', 'create');
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
   }
 
-  function remove(id) {
-    initialise('action', 'remove', id);
+  function toggleFilter() {
+    vm.filterEnabled = !vm.filterEnabled;
+    vm.gridOptions.enableFiltering = vm.filterEnabled;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
   }
 
-  function cancel() {
-    initialise('default');
-  }
-
-  function submit(actionForm) {
-    if (actionForm.$invalid) {
-      return;
-    }
-    map[vm.action].submit(vm.depot.uuid);
-  }
+  function loadDepots() {
+    vm.loading = true;
 
-  function createDepot() {
-    vm.depot.enterprise_id = SessionService.enterprise.id;
-    DepotService.create(vm.depot)
-    .then(function (res) {
-      vm.state.created = true;
-      vm.view = 'success';
-    })
-    .then(depotsList)
-    .catch(errorHandler);
-  }
-
-  function updateDepot(uuid) {
-    DepotService.update(uuid, vm.depot)
-    .then(function (res) {
-      vm.state.updated = true;
-      vm.view = 'success';
-    })
-    .then(depotsList)
-    .catch(errorHandler);
-  }
-
-  function removeDepot(uuid) {
-    DepotService.delete(uuid)
-    .then(function (res) {
-      vm.state.removed = true;
-      vm.view = 'success';
+    Depots.read(null, { full : 1 })
+    .then(function (data) {
+      vm.gridOptions.data = data;
     })
-    .then(depotsList)
-    .catch(errorHandler);
+    .catch(Notify.handleError)
+    .finally(function () {
+      vm.loading = false;
+    });
   }
 
-  function errorHandler(err) {
-    vm.state.errored = true;
-    console.log(err);
+  // switch to delete warning mode
+  function deleteDepot(depot) {
+    ModalService.confirm('FORM.DIALOGS.CONFIRM_DELETE')
+    .then(function (bool) {
+      if (!bool) { return; }
+
+      Depots.delete(depot.uuid)
+      .then(function () {
+        Notify.success('DEPOT.DELETED');
+        loadDepots();
+      })
+      .catch(Notify.handleError);
+    });
   }
 
-  function initialise(view, action, uuid) {
-    vm.state.reset();
-    vm.view   = view;
-    vm.action = action;
-    vm.depot  = {};
-
-    vm.actionTitle = map[action].title;
-
-    if (uuid && (action === 'update' || action === 'remove')) {
-      DepotService.read(uuid)
-      .then(function (depot) {
-        vm.depot = depot;
-      });
-    }
+  // update an existing depot
+  function editDepot(depotObject) {
+    $state.go('depots.edit', { depot : depotObject });
   }
 
-  function refreshValidation() {
-    vm.state.errored = vm.depot.text ? false : true;
+  // create a new depot
+  function createDepot() {
```

Comment:
Instead of having a `createDepot()` function, you could easily just have a `<a ui-sref="depots.create"></a>`.  It would reduce the amount of JS code ... but I'm not sure the best practice here.